### DISCUSSION TITLE
AttributeError: 'Window' object has no attribute 'anvilLabs' at app/anvil_labs/web_worker.py, line 69

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-labs#127] [Q&A]
Hi Team
We have just started seeing this error pop up in our logs.

`AttributeError: 'Window' object has no attribute 'anvilLabs'
at app/anvil_labs/web_worker.py, line 69`

It is coming from our production code and unfortunately occurs before I can print any useful information to the logs.
I don't see any changes in the code and I also cannot replicate it.. Any ideas?

**Comment by s-cork** [comment]
No changes to the code from our end - is it a specific user?
Or all users?
If it's a specific user we'd want to know what browser/device they're using

**Comment by rhurlbatt** [comment]
Ok so I finally have some more detail on this.
The problem is device specific and the one we have found that is having it is an iPhone 6 (#facepalm) running version 12.5.7
Does this give us any more insight?

**Reply by s-cork** [reply]
Possibly. That's now an old version of iOS - 5 years old or so. So it's likely we just won't be able to support it with the current implementation. I suggest your user upgrades their iOS or their phone 😉. 
But if they're not ready to spend the dollars it's likely there's some piece of modern web syntax that breaks in iOS 12.x.
I have access to browser stack so I can probably try to run an old version of safari and see if the issue arises there too. Depending on the triviality of the issue we can determine how easy the fix is (and whether it's worth it).

**Reply by s-cork** [reply]
Ok so it's just a syntax thing - I changed the javascript build to be es2019 which should be supported by ios 12.x
Tested on safari 12.x and can confirm I could reproduce and that the error is now fixed

**Reply by rhurlbatt** [reply]
Ok wicked. - Thank you Stu!  I will get in contact with the user and confirm it is fixed his end.
